### PR TITLE
Add auto theme option for detecting the theme from system/browser preferences

### PR DESCRIPTION
**Pull request by remmycat**
- I refactored the theme code a little and expanded some comments.
- For now I took the normal "theme" value out of the context, as it would be bug-prone to read from it in normal components, it would lead to hydration issues (or flickering when waiting for mount first).

### CODE REVIEW

**Comment by Sendouc** [comment]
Hey! Looking nice 🚀  I'll take a proper look later but firstly is there maybe a small bug?

https://user-images.githubusercontent.com/38327916/199049324-077b0d50-98f2-4450-8900-a76cc1550450.mov

Seems like when I'm updating my OS preference the affect on sendou ink is inversed?

**Comment by remmycat** [comment]
Haha, nice find, will fix it! I was trying this earlier and thought the browser just doesn't trigger the event 🤦

**Comment by remmycat** [comment]
Should be fixed now, I just had the ternary flipped accidentally.

**Comment by remmycat** [comment]
Fixed a comment

File: app/components/layout/LanguageChanger.tsx [hunk 1]
```diff
@@ -5,11 +5,16 @@ import { GlobeIcon } from "../icons/Globe";
...
 export function LanguageChanger() {
-  const { i18n } = useTranslation();
+  const { t, i18n } = useTranslation();
 
   return (
     <Popover
-      buttonChildren={<GlobeIcon className="layout__header__button__icon" />}
+      buttonChildren={
+        <GlobeIcon
+          alt={t("header.language")}
```
Comment (Sendouc) [inline app/components/layout/LanguageChanger.tsx:14]: Nice that you added these!

File: app/components/layout/ThemeChanger.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          const Icon = ThemeIcons[theme];
+          const selected = userTheme === theme;
+          return (
+            <Button
+              variant="minimal"
+              key={theme}
+              tiny
+              icon={<Icon alt="" />}
+              // TODO: Remove this and find better semantic representation than
+              // just multiple buttons. Maybe radio group?
```
Comment (Sendouc) [inline app/components/layout/ThemeChanger.tsx:46]: Maybe https://headlessui.com/react/radio-group (for future)

**Review by Sendouc — APPROVED** [review]
Really good work 👌 Good to be merged imo.

**Comment by remmycat** [comment]
Finally got around to fixing the conflict and merging it 🙂

**Pull request merged**